### PR TITLE
fix: allow for of loops

### PR DESCRIPTION
**Pull request by linaGirl**
I can't figure what's not OK with them in some cases (like when working on async code). Airbnbs argumentation is about immutability. Mine is about readability and ease of use. I like iterators, they are great! See https://github.com/airbnb/javascript#iterators-and-generators and https://github.com/airbnb/javascript/issues/1122#issuecomment-259876436

### CODE REVIEW

**Comment by fxstr** [comment]
Suggestion is okay with me and definitly makes sense for async operations. It's up to the programmer to use and up to the reeviwer to ensure the (better fitting) Array methods whenever possible. Please inform the team.

**Review by fxstr — APPROVED** [review]

**Comment by linaGirl** [comment]
i'd like to add the [`"no-await-in-loop": "off"`](https://eslint.org/docs/rules/no-await-in-loop) rule so that i actually can use async operations inside of a loop. What's your take on that @fxstr ?

**Review by rcky — APPROVED** [review]
I used them and ignored the linter :D In my opinion it is the most readable way to do sequential async operations  (you just have to know when they are appropriate, and when they're not).

**Comment by fxstr** [comment]
@eventEmitter Should we update https://github.com/joinbox/guidelines?

File: rules/joinbox-custom.js [hunk 1]
```diff
@@ -19,5 +19,9 @@ module.exports = {
         'class-methods-use-this': 'off',
         // don't require an extensions for imports from packages
         'import/extensions': 'ignorePackages',
+        // allow for of loops
```
Comment (DominicTobias) [inline rules/joinbox-custom.js]: +1, what a stupid rule, especially in Node env

**Pull request merged**